ui/lib/kv/addon/components/page/secret/metadata-edit.hbs

```diff
@@ -1,6 +1,63 @@
-<KvPageHeader @breadcrumbs={{@breadcrumbs}} @model={{@model}} @pageTitle="Edit Secret Metadata" />
+<KvPageHeader @breadcrumbs={{@breadcrumbs}} @model={{@metadata}} @pageTitle="Edit Secret Metadata" />
 
 <hr class="is-marginless has-background-gray-200" />
-<p class="has-top-margin-m">
+<p class="has-top-margin-m has-bottom-margin-m">
   The options below are all version-agnostic; they apply to all versions of this secret.
-</p>
\ No newline at end of file
+</p>
+{{#if @metadata.canUpdateMetadata}}
+  <form {{on "submit" (perform this.save)}}>
+    <div class="box is-sideless is-fullwidth is-marginless">
+      <NamespaceReminder @mode="update" @noun="KV secret metadata" />
+      <MessageError @errorMessage={{this.errorBanner}} class="has-top-margin-s" />
+      {{#each @metadata.formFields as |attr|}}
+        {{#if (eq attr.name "customMetadata")}}
+          <FormField @attr={{attr}} @model={{@metadata}} @modelValidations={{this.modelValidations}} />
+          <label class="title has-top-padding-m is-4">
+            Additional options
+          </label>
+        {{else if (not-eq attr.name "customMetadata")}}
+          <FormField @attr={{attr}} @model={{@metadata}} @modelValidations={{this.modelValidations}} />
+        {{/if}}
+      {{/each}}
+    </div>
+    <div class="field is-grouped is-grouped-split is-fullwidth box is-bottomless">
+      <div class="has-top-padding-s">
+        <button
+          type="submit"
+          class="button is-primary {{if this.save.isRunning 'is-loading'}}"
+          disabled={{this.save.isRunning}}
+          data-test-kv-metadata-save
+        >
+          Update
```

Comment:
In the old code this said Save, not sure if changing here was intentional or not

Reply:
Yeah, I changed it. We're really updating it, so if felt appropriate.